### PR TITLE
endpoint to run package analyzer jobs

### PR DESCRIPTION
**Pull request by bissenbay**
Depends-On: https://github.com/thoth-station/common/pull/443

Endpoint to handle package analyzer jobs

### CODE REVIEW

File: openapi/openapi.yaml [hunk 1]
```diff
@@ -969,3 +969,62 @@ paths:
...
+            default: https://pypi.org/simple
+        - name: debug
+          in: query
+          required: false
+          description: Be verbose about what's going on.
+          schema:
+            type: boolean
+            default: false
+      responses:
+        "202":
```
Comment (bissenbay) [inline openapi/openapi.yaml]: Do I need to handle other responses as well?

File: thoth/management_api/api_v1.py [hunk 2]
```diff
@@ -448,3 +448,10 @@ def _do_schedule(parameters: dict, runner: typing.Callable, **runner_kwargs):
         },
         202,
     )
+
+
+def analyze_package(package_name: str, package_version: str, index_url: str, debug: bool):
+    """Fetch digests for packages in Python ecosystem."""
+    parameters = locals()
+
+    return _do_schedule(parameters, _OPENSHIFT.schedule_package_analyzer, output=Configuration.THOTH_ANALYZER_OUTPUT)
```
Comment (bissenbay) [inline thoth/management_api/api_v1.py]: Should I request for the output from the user?

File: openapi/openapi.yaml (hunk 1 above)
Comment (fridex) [reply openapi/openapi.yaml]: I think you can keep it as is.

File: openapi/openapi.yaml [hunk 3]
```diff
@@ -969,3 +969,62 @@ paths:
...
+                properties:
+                  analysis_id:
+                    type: string
+                    description: >
+                      An id of submitted analysis for checking its status and its results.
+                  parameters:
+                    type: object
+                    description: >
+                      Parameters echoed back to user (with default parameters if omitted).
+                  cached:
```
Comment (fridex) [inline openapi/openapi.yaml]: I don't think we will cache these results.

File: thoth/management_api/api_v1.py (hunk 2 above)
Comment (fridex) [reply thoth/management_api/api_v1.py]: No, it will be always result-api, the code is correct.

File: thoth/management_api/configuration.py [hunk 4]
```diff
@@ -43,6 +43,7 @@ class Configuration:
     THOTH_MANAGEMENT_API_TOKEN = os.environ['THOTH_MANAGEMENT_API_TOKEN']
     THOTH_MIDDLETIER_NAMESPACE = os.environ['THOTH_MIDDLETIER_NAMESPACE']
     THOTH_SOLVER_SUBGRAPH_CHECK_API = THOTH_RESULT_API_URL + '/api/v1/subgraph-check'
+    THOTH_ANALYZER_OUTPUT = THOTH_RESULT_API_URL + '/api/v1/package-analysis-result'
```
Comment (fridex) [inline thoth/management_api/configuration.py]: Be more specific here:

```diff
- THOTH_ANALYZER_OUTPUT
+ THOTH_PACKAGE_ANALYZER_OUTPUT
```

File: openapi/openapi.yaml [hunk 5]
```diff
@@ -969,3 +969,57 @@ paths:
           description: On invalid request.
         "401":
           description: On invalid secret.
+
+  /package-analyzer:
+    post:
+      tags: [Package Analyzer]
+      x-openapi-router-controller: thoth.management_api.api_v1
+      operationId: analyze_package
+      summary: Gather digests of packages and files present inside packages.
```
Comment (fridex) [inline openapi/openapi.yaml]: This description is misleading - it gathers digests from packaged Python packages.

File: openapi/openapi.yaml [hunk 6]
```diff
@@ -969,3 +969,57 @@ paths:
...
+        - name: package_version
+          in: query
+          required: true
+          description: Package version for which digests should be fetched.
+          schema:
+            type: string
+        - name: index_url
+          in: query
+          required: false
+          description: A comma separated list of Python package indexes (package sources) to gather digests from.
```
Comment (fridex) [inline openapi/openapi.yaml]: Let's keep it simple - just one `index_url` per run (no comma-separated list).

File: openapi/openapi.yaml [hunk 7]
```diff
@@ -969,3 +969,57 @@ paths:
...
+          schema:
+            type: string
+            default: https://pypi.org/simple
+        - name: debug
+          in: query
+          required: false
```
Comment (fridex) [inline openapi/openapi.yaml]: You state here it's not required, but the function signature does not have any default parameter. Will it work? (I'm not sure, never tried it like that).

Comment (bissenbay) [reply openapi/openapi.yaml]: @fridex It's already deployed, do you want to take a look at http://management.test.thoth-station.ninja/api/v1/ui/#/?

Comment (fridex) [reply openapi/openapi.yaml]: OK, I will trust you :wink:

Comment (bissenbay) [reply openapi/openapi.yaml]: It's better for me to fix it earlier, of course, I took a look at this example: https://github.com/thoth-station/management-api/blob/master/openapi/openapi.yaml#L30

**Pull request merged**